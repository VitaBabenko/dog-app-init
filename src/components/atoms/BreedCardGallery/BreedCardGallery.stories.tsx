import type { Meta, StoryObj } from '@storybook/react';
import { BreedCardGallery } from '.';

const meta = {
  title: 'Atoms/BreedCardGallery',
  component: BreedCardGallery
} satisfies Meta<typeof BreedCardGallery>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    id: 1,
    image: {
      url: ''
    },
    name: '',
    temperament: ''
  }
};
