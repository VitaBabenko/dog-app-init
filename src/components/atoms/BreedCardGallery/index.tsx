import { Button, Typography } from '@mui/material';
import { FC, useState } from 'react';
import { Link } from 'react-router-dom';
import { CardWrapper } from '../CardWrapper';

type BreedCardGalleryProps = {
  id: number;
  image: {
    url: string;
  };
  name: string;
  temperament: string;
};

export const BreedCardGallery: FC<BreedCardGalleryProps> = ({
  id,
  image,
  name,
  temperament
}) => {
  const [isHoveredCard, setIsHoveredCard] = useState<boolean>(false);

  const handleMouseOver = () => setIsHoveredCard(true);
  const handleMouseOut = () => setIsHoveredCard(false);

  return (
    <CardWrapper
      key={id}
      variant="primary"
      borderRadius={0}
      sx={{ paddingBottom: '29px' }}
      onMouseOver={handleMouseOver}
      onMouseOut={handleMouseOut}
    >
      <img src={image.url} alt={name} />
      <Typography fontSize={26} color="#5F556D">
        {name}
      </Typography>
      <Typography>
        <Typography component="span" fontSize={20} color="#5F556D">
          Temperament:
        </Typography>
        <Typography component="span" fontSize={20} color="#ADA7B8">
          {temperament}
        </Typography>
      </Typography>
      {isHoveredCard && (
        <Link to={`${id}`}>
          <Button variant="text" color="inherit">
            more
          </Button>
        </Link>
      )}
    </CardWrapper>
  );
};
