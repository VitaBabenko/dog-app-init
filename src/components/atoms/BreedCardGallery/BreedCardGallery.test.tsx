import { composeStories } from '@storybook/react';
import * as stories from './BreedCardGallery.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('BreedCardGallery', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
