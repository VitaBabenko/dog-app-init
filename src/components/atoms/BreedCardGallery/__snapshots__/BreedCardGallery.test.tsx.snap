// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedCardGallery should match snapshot Component 1`] = `
.emotion-0 {
  border-radius: 0;
  position: relative;
  cursor: pointer;
  -webkit-background-position: center;
  background-position: center;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  overflow: hidden;
  max-width: 590px;
  min-height: 90px;
  background-color: #FFF;
  box-shadow: 6px 6px 0px 0px #000000;
  border: 1px solid #090808;
  border-radius: 20px;
  padding-bottom: 29px;
}

.emotion-0 img {
  display: block;
  width: 100%;
  object-fit: cover;
}

.emotion-0:hover,
.emotion-0:active {
  box-shadow: 6px 6px 8px 0px #921FED;
}

.emotion-1 {
  margin: 0;
  font-family: 'Dosis',sans-serif;
  font-weight: 400;
  font-size: 1.4285714285714286rem;
  line-height: 1.5;
  font-size: 26px;
  color: #5F556D;
}

.emotion-2 {
  margin: 0;
  font-family: 'Dosis',sans-serif;
  font-weight: 400;
  font-size: 1.4285714285714286rem;
  line-height: 1.5;
}

.emotion-3 {
  margin: 0;
  font-family: 'Dosis',sans-serif;
  font-weight: 400;
  font-size: 1.4285714285714286rem;
  line-height: 1.5;
  font-size: 20px;
  color: #5F556D;
}

.emotion-4 {
  margin: 0;
  font-family: 'Dosis',sans-serif;
  font-weight: 400;
  font-size: 1.4285714285714286rem;
  line-height: 1.5;
  font-size: 20px;
  color: #ADA7B8;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    variant="primary"
  >
    <img
      alt=""
      src=""
    />
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-2"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 emotion-3"
      >
        Temperament:
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 emotion-4"
      />
    </p>
  </div>
</div>
`;
